File: BE/health-service/src/main/java/com/s206/health/exercise/dto/response/DailyExerciseStatsResponse.java
package com.s206.health.exercise.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DailyExerciseStatsResponse {
    private List<DailyExercise> dailyExercises;

    @Getter
    @Builder
    public static class DailyExercise {
        private LocalDate date;
        private Integer totalCalories;
        private Integer exerciseCount;
        private List<ExerciseDetail> exercises;
    }

    @Getter
    @Builder
    public static class ExerciseDetail {
        private Integer exerciseId;
        private String exerciseName;
        private String categoryName;
        private Integer exerciseTime;
        private Integer exerciseCalorie;
    }
}

File: FE/mobile/src/main/java/com/example/diaviseo/network/exercise/dto/res/DailyExerciseStatsResponse.kt
package com.example.diaviseo.network.exercise.dto.res

data class DailyExerciseStatsResponse(
    val dailyExercises: List<DailyExercise>
) {
    data class DailyExercise(
        val date: String,
        val totalCalories: Int,
        val exerciseCount: Int,
        val exercises: List<ExerciseDetail>
    )

    data class ExerciseDetail(
        val exerciseId: Int,
        val exerciseName: String,
        val categoryName: String,
        val exerciseTime: Int,
        val exerciseCalorie: Int
    )
}